File: source/java/formationds/external/model/src/main/java/com/formationds/client/model/Role.java
/**
 * Copyright (c) 2015 Formation Data Systems.  All rights reserved.
 */

package com.formationds.client.model;

import java.util.ArrayList;
import java.util.EnumSet;
import java.util.List;

public enum Role implements RoleDescriptor {

    ADMIN(Feature.SYS_MGMT, Feature.TENANT_MGMT, Feature.USER_MGMT, Feature.VOL_MGMT),
    TENANT_ADMIN(Feature.TENANT_MGMT, Feature.USER_MGMT, Feature.VOL_MGMT),
    USER(Feature.VOL_MGMT, Feature.USER_MGMT);

    private final EnumSet<Feature> features;

    Role( Feature f1, Feature... features ) {
        this.features = EnumSet.of( f1, features );
    }

    @Override
    public List<Feature> getFeatures() {
        return new ArrayList<>( features );
    }
}

File: source/cli/src/fds/model/admin/role.py
class Role(object):
    '''
    Created on May 29, 2015
    
    @author: nate
    '''

    def __init__(self, name="TENANT_ADMIN", features=["VOL_MGMT", "USER_MGMT"] ):
        self.name = name
        self.features = features
        
    @property
    def name(self):
        return self.__name
    
    @name.setter
    def name(self, name):
        
        if name not in ("TENANT_ADMIN", "ADMIN", "USER"):
            raise TypeError()
        
        self.__name = name
        
    @property
    def features(self):
        return self.__features
    
    @features.setter
    def features(self, features):
        
        self.__features = features